jobsdb/jobsdb.go

```diff
@@ -834,6 +834,16 @@ func (jd *HandleT) init() {
 		jd.dbHandle = sqlDB
 	}
 	jd.workersAndAuxSetup()
+
+	// Database schema migration should happen early, even before jobsdb is started,
+	// so that we can be sure that all the necessary tables are created and considered to be in
+	// the latest schema version, before rudder-migrator starts introducing new tables.
+	jd.dsListLock.WithLock(func(l lock.LockToken) {
+		switch jd.ownerType {
+		case Write, ReadWrite:
+			jd.setupDatabaseTables(l, jd.clearAll)
```

Comment:
**_Note:_** all gateways will need to be upgraded to the latest version so that the necessary schema migration will performed on `gw` datasets.